fuse/fuse.cpp

```diff
@@ -167,22 +167,39 @@ int main(int argc, char *argv[])
 		if (stdin.empty()) stdin = "/dev/null";
 
 #ifdef __linux__
-		// Allow overriding this to; e.g /build
-		std::string workspace = cwd;
-
 		uid_t real_euid = geteuid();
 		gid_t real_egid = getegid();
 
+		// Allow overriding this to; e.g /mnt
+		std::string workspace = cwd;
+		uid_t euid = real_euid;
+		gid_t egid = real_egid;
+		int flags = CLONE_NEWNS|CLONE_NEWUSER;
+
+		for (auto &res : jast.get("resources").children) {
+		  std::string &key = res.second.value;
+		  if (key == "isolate/workspace") workspace = "/mnt";
```

Comment:
I think the path shouldn't be directly on top of a common mount point.
Perhaps something like `/mnt/wake` or `/tmp/wake`? 

Not that I think it's likely, but someone might have some toolchain (via a Resource) that was in `/mnt` that is now hidden

Reply:
> If we do this it's likely a good time to flip the default on `PATH=/usr/bin:/bin` too, since there'd be plenty of Plans to manually edit

Definitely.

Yeah. Your proposal to deal with /tmp/wake existing was to mount tmpfs over /tmp before doing a mkdir(/tmp/wake). That then hides everything in /tmp, similar to how this hides everything in /mnt.

What I would really like to do is mkdir(/build), but AFAIK there is no way to fake that.

Reply:
Hmm, hiding /tmp could be an advantage

We could also do an "if /build exists use it instead"

Reply:
If you are allowed to mount an overlay on top of `/` you could create `/build`

Reply:
Looking through the FHS, the best defined candidates to hide are:
- /boot (probably ok to hide)
- /mnt (Temporarily mounted filesystems)

Here is another option:
/var/cache/wake
... and make that directory as part of wake installation.

What's a bit unfortunate is that /var is not under /usr, so it's always a pain to work with when you have a relocatable package like wake that can live in /usr/local/...

I would propose to use /var/cache/wake (regardless of where wake is installed).

Reply:
Using BINPATH/../lib/wake/build might be a better choice, but it's not exactly the intended FHS for /usr/lib.

Reply:
Building on your (if it exists idea), how about this policy:
Use /var/cache/wake if it exists
Otherwise, use BINPATH/../lib/wake/build and create that as part of the wake install.

Reply:
> ... and make that directory as part of wake installation.

We could do that tactic with `/build` (or perhaps `/wake`, much like Nix does `/nix`), if they're using deb/rpm to install then they have root to make that directory.

Reply:
I don't think `/var/cache/wake` handles the case where wake was compiled from source by a non-root user, who hasn't previously installed wake before as root

Reply:
Ubuntu packages are not allowed to create directories in the top of the filesystem (/build), so /var/cache/wake is a better option since that is FHS compliant and will work on all distros.

I agree that you may not have permission to create /var/cache/wake (or /build or /wake), so we need a fallback. Since a non-root user can pick where they install wake, I've used a path relative to the users selected installation location as the fallback.